hedera-node/src/main/java/com/hedera/services/state/migration/MigrationRecordsManager.java

```diff
@@ -124,6 +143,36 @@ private TransactionBody.Builder synthStakingFundCreate(final long autoRenewPerio
 		return TransactionBody.newBuilder().setCryptoCreateAccount(txnBody);
 	}
 
+	private void publishContractFreeAutoRenewalRecords(final Instant now) {
+		final var changedContracts = new HashMap<EntityNum, Long>();
+		accounts.get().forEach((a, b) -> {
```

Comment:
⚠️ `MerkleMap` still doesn't override `Map.forEach()`, so this iterator is extremely inefficient. We should use the [`MiscUtils.forEach()` helper](https://github.com/hashgraph/hedera-services/blob/master/hedera-node/src/main/java/com/hedera/services/utils/MiscUtils.java#L846) that delegates to `MerkleMap.forEachNode()`.

Reply:
As discussed, `MerkleMap.forEach` since that seems to be better performant. Reverted back to it.